### ISSUE TITLE
Python stops the program when building a template or  converting a SBML file

### ISSUE THREAD

**Issue by PolTorrentTornos** [SystemsBioinformatics/cbmpy-metadraft#8]
[Hello!]
[L casei 12A.zip](https://github.com/SystemsBioinformatics/cbmpy-metadraft/files/6228555/L.casei.12A.zip)

I'm new at MetaDraft. I was trying to build my own template by:
1-convert my SBML level 2 to level 3 FBC v3
2-create a template with a Genbank file + my converted SBML file. 

In both of these steps anaconda finds an "Unhandled Python exception" and shuts down MetaDraft. I'm using the anaconda prompt with python v3.8.5. Interestingly, after step 1 I did receive a converted SBML and after step 2 (using the converted SBML from step 1) I received a .xlsx file. However my template is not shown in the "select the target network(s)" window. 

Any idea how to solve this? 
Best, 

1st update:the same problem happens with other files. 
2nd update: I tried using python 3.6. I then managed to upgrade the sbml version and to build templates, but didn't manage to create metaproteome, python shutted down again. I tried installing python 2.7 and cbmpy 0.7.5 (that only needs biopython 1.7.6., more recent versions would require a more updated python) but then libsbml was missing. I could only install libsbml with "pip install python-libsbml-experimental" .
 
Now the error is that no QtGui is defined, which I solved by  installing  PyQT4. However then there is a conflict with PyQT5, which makes me think that the QtGui was not missing. Any ideas how to proceed?

**Comment by AdityaVaze** [comment]
Hello, did you ever manage to fix this? I'm encountering the same problem. MetaDraft passes the systemtest requirements, but when I try to build a template model, it hangs and then crashes with the "2 Unhandled Python Exception" error message.

**Comment by bgoli** [comment]
Can you try the following and see if it solves the problem:

Clone the repository:

```
git clone https://github.com/SystemsBioinformatics/cbmpy-metadraft.git
```

Change into the MetaDraft directory and activate the template submodule:

```
cd cbmpy-metadraft
git submodule update --init --remote -- modeldb/2019-1/
```

Creating the environment, this should work on all operating systems. Open a terminal and type:

```
conda env create -f environment.yml
conda activate metadraft3
```

One Windows run the system test (optional) and start MetaDraft:
```
runwin.bat
```

**Comment by AdityaVaze** [comment]
This is how I initially set up the system, do you mean I should do a full reinstall?

**Comment by bgoli** [comment]
The problems you are experiences have got to do with changes in the way Qt packages are supplied. You should be able to upgrade by doing the following (in your metadraft directory)

```
git pull
```
edit **environment.yml** and change 
```
name: metadraft3
```
to
```
name: metadraft4
```
then do 
```
conda env create -f environment.yml
conda activate metadraft4
python systemtest.py
```
and finally
```
python metadraft.py
```

HTH

**Comment by AdityaVaze** [comment]
This managed to get it started, thanks! The template model building doesn't work yet, will try to figure out why tomorrow. I'm using the human model Recon3D.xml downloaded from the BiGG database and the human genome CRCh38.p13.gbff from Genbank.

**Comment by bgoli** [comment]
I included some test files to test the template creation function in `testfiles\createtemplate`. Note that both the *.gb* and the *.xml* files must be in the same directory to work and that gene id's need to be compatible - let me know if you need help with this.

**Comment by AdityaVaze** [comment]
Hello again! I tried running the test files and it seems to identify the genes correctly but crashes while writing the model with the following error:

```
Traceback (most recent call last):
  File "C:\Users\vazea\Documents\Thesis\MetaDraft\cbmpy-metadraft\libpython\qtmetadraft.py", line 1920, in bp_createSeqPlusProcess
    ret = biotools.createSeqplusModel(
  File "C:\Users\vazea\Documents\Thesis\MetaDraft\cbmpy-metadraft\libpython\biotools.py", line 1033, in createSeqplusModel
    cbmpy.writeSBML3FBCV2(
  File "C:\Users\vazea\Anaconda3\envs\metadraft4\lib\site-packages\cbmpy\CBWrite.py", line 195, in writeSBML3FBCV2
    return CBXML.sbml_writeSBML3FBC(
  File "C:\Users\vazea\Anaconda3\envs\metadraft4\lib\site-packages\cbmpy\CBXML.py", line 2875, in sbml_writeSBML3FBC
    sbml_setDescription(cs3.model, fba)
  File "C:\Users\vazea\Anaconda3\envs\metadraft4\lib\site-packages\cbmpy\CBXML.py", line 1133, in sbml_setDescription
    annoSTRnew = sbml_writeKeyValueDataAnnotation(fba.getAnnotations())
  File "C:\Users\vazea\Anaconda3\envs\metadraft4\lib\site-packages\cbmpy\CBXML.py", line 2165, in sbml_writeKeyValueDataAnnotation
    Kval = cgi.escape(str(annotations[K]))
AttributeError: module 'cgi' has no attribute 'escape'
```
With Recon3D this issue also occurs, but there seems to be an additional problem with the gene ids. Which is strange, because Recon3D is itself based on exactly the genome file that I'm using.

Would it be more appropriate to continue the discussion elsewhere? Shall I open a new issue? I'm not too familiar with the Git etiquette. Thanks for your help!

**Comment by bgoli** [comment]
Thanks, as we will be discussing project related details, could you [email](https://research.vu.nl/en/persons/bg-olivier) me directly.
